### ISSUE TITLE
openssl@1.1: Can't locate Test/Harness.pm in @INC on CentOS 8

### ISSUE THREAD

**Issue by rusdevops** [Homebrew/linuxbrew-core#20079]
formula: openssl@1.1/1.1.1f
plarform: CentOS 8

```
brew install ag
Installing the_silver_searcher dependency: openssl@1.1
Can't locate Test/Harness.pm in @INC (you may need to install the Test::Harness module)
```

full log: https://github.com/fast-ide/bootstrap/runs/578734027

**Comment by alanhoyle** [comment]
I am also seeing this on CentOS Linux release 7.7.1908 (Core).

**Comment by iMichka** [comment]
You can add a resource block with the needed dependency to the formula.

Grep for "Xml::Parser" to see how it is done in other formulae. A pull request to fix this would be welcome.

**Comment by lewis6991** [comment]
Seeing this too on RHE7

**Comment by rusdevops** [comment]
@iMichka it looks like you need to add `perl -MCPAN -e 'install Test::Harness'`, please see:
https://techglimpse.com/cant-locate-testharness-pm-perl-error-solution/

**Comment by alanhoyle** [comment]
Based on the XML::Parser suggestion, I think this is what @iMichka is suggesting be added to `./linuxbrew-core/Formula/openssl@1.1.rb`:

```
  resource "Test::Harness" do
    url "https://cpan.metacpan.org/authors/id/L/LE/LEONT/Test-Harness-3.42.tar.gz"
    sha256 "0fd90d4efea82d6e262e6933759e85d27cbcfa4091b14bf4042ae20bab528e53"
  end
```

I don't know how to test this to see if it would work.  would I be able to edit in place the following file? 
`~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl@1.1.rb`

**Comment by issyl0** [comment]
Yep: `brew edit openssl@1.1`, then `brew reinstall openssl@1.1`. Let us know how you get on!

**Comment by alansill** [comment]
I ran into this on CentOS 7 also. Tried the approach mentioned above. It wasn't clear where to put this section into the file when doing the brew edit step, so I put it into the section under ```unless OS.mac?``` as follows:
```
  unless OS.mac?
    resource "cacert" do
      # homepage "http://curl.haxx.se/docs/caextract.html"
      url "https://curl.haxx.se/ca/cacert-2020-01-01.pem"
      mirror "https://gist.githubusercontent.com/dawidd6/16d94180a019f31fd31bc679365387bc/raw/ef02c78b9d6427585d756528964d18a2b9e318f7/cacert-2020-01-01.pem"
      sha256 "adf770dfd574a0d6026bfaa270cb6879b063957177a991d453ff1d302c02081f"
    end
    # See https://github.com/Homebrew/linuxbrew-core/issues/20079
    resource "Test::Harness" do
      url "https://cpan.metacpan.org/authors/id/L/LE/LEONT/Test-Harness-3.42.tar.gz"
      sha256 "0fd90d4efea82d6e262e6933759e85d27cbcfa4091b14bf4042ae20bab528e53"
    end
  end
```
The results however were the same after doing this and issuing the ```brew reinstall openssl@1.1``` command.

I also tried doing a ```cpan -i Test::Harness``` externally before ```brew reinstall openssl@1.1``` but this didn't help either.

**Comment by alanhoyle** [comment]
I also tried putting the resource  section in a few places by messing with brew edit openssl@1.1 and I was unable to figure it out either.  I'm a complete neophyte with brew formulae, and I'm sure an experienced developer could do this is a second.  This is especially frustrating as the only reason why I'm running into this issue is because openssl is apparently a second- or third-degree dependency on the actual tool I want to use (csvkit), and it works fine on my mac installed with brew.  

I am vaguely amused that there's another person named "Alan" who is running into the same issue on the same OS, and trying what appear to be very similar things with the same level of experience.

**Comment by rusdevops** [comment]
workaround: https://github.com/fast-ide/linuxbrew/blob/e0b04fe10325d66d8b715b63bcc3e604231de040/scripts/redhat/requirements.sh#L10-L12

**Comment by iMichka** [comment]
I opened a pull request to fix this: #20243
Can you test this change?

**Comment by alansill** [comment]
This worked:

> workaround: https://github.com/fast-ide/linuxbrew/blob/e0b04fe10325d66d8b715b63bcc3e604231de040/scripts/redhat/requirements.sh#L10-L12

**Comment by iMichka** [comment]
Fixed by #20243
